src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,153 +193,164 @@ public boolean canBeRotated() {
+    {
 
-                if (itemIn.stackSize - 1 <= 0) {
-                    itemOut = itemIn.copy();
-                    itemIn = null;
-                } else {
-                    itemOut = itemIn.copy();
 
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
+        //checking for the machine type
+        if (machineTier == null)
+            machineTier = MachineTier.byMeta(getBlockMetadata());
 
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
+        if (!canWork()) {
+            machineActive = false;
+            this.markForUpdate();
+        }
 
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
+        if (ticksRemaining == 0 && inventory.getStackInSlot(0) == null && machineTier != MachineTier.TIER_0) {
+            machineActive = false;
+        }
 
-                ticksRemaining = 200;
+        if (this.canWork()) {
+            if (machineTier == MachineTier.TIER_0) {
+                fuelRemaining--;
+                machineActive = fuelRemaining > 0;
+            } else if (ticksRemaining > 0) {
+                container.takePower(powerUsage, false);
                 machineActive = true;
+            }
+            if (machineActive && ticksRemaining > 0) {
+                ticksRemaining--;
+            }
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-                this.markForUpdate();
-                this.markDirty();
+        if (fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0) {
+            if (inventory.getStackInSlot(1) != null) {
+                burnTime(1);
             }
+            if (inventory.getStackInSlot(0) != null) {
+                burnTime(0);
+            }
+        }
 
 
-            if (fuelRemaining == 0 && machineActive && machineTier == machineTier.TIER_0) {
-                ticksRemaining = 0;
-                test = false;
+        if (this.canWork() && inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
+            if (!cancrush(inventory.getStackInSlot(0)))
+                return;
 
+            ItemStack itemIn = inventory.getStackInSlot(0);
+            ItemStack itemOut;
+
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
             } else {
-                test = false;
+                itemOut = itemIn.copy();
+                itemOut.stackSize = 1;
+                itemIn.stackSize = itemIn.stackSize - 1;
             }
+            if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
+            if (itemOut.stackSize == 0) itemOut = null;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)